packages/wonder-blocks-dropdown/src/components/dropdown-core.js

```diff
@@ -579,11 +599,98 @@ class DropdownCore extends React.Component<Props, State> {
         return null;
     }
 
+    /**
+     * Handles click events for each item in the dropdown.
+     */
+    handleItemClick: (focusIndex: number, item: DropdownItem) => void = (
+        focusIndex: number,
+        item: DropdownItem,
+    ) => {
+        this.handleClickFocus(focusIndex);
+        if (item.component.props.onClick) {
+            item.component.props.onClick();
+        }
+        if (item.populatedProps.onClick) {
+            item.populatedProps.onClick();
+        }
+    };
+
+    /**
+     * Determines which rendering strategy we are going to apply to the options
+     * list.
+     */
+    shouldVirtualizeList(): boolean {
+        // Verify if the list is long enough to be virtualized (passes the
+        // threshold).
+        return this.props.items.length > VIRTUALIZE_THRESHOLD;
+    }
+
+    /**
+     * Renders the non-virtualized list of items.
+     */
+    renderList(): React.Node {
+        let focusCounter = 0;
+        const itemRole = this.getItemRole();
+
+        // if we don't need to virtualize, we can render the list directly
+        return this.props.items.map((item, index) => {
```

Comment:
How much of this logic is shared with dropdown-core-virtualized.js?

Reply:
It's like 60% - 70%. The virtualized version process this in two steps:

1. Modifies the items in this file (`parseVirtualizedItems()`), so we can pass any logic related to keyboard/focus navigation. (I added a note in that function explaining that distinction).
2. Then, in `dropdown-core-virtualized-item`, it maps every item to clone it using the data processed in point # 1.

The "regular" version does both steps directly here, so there are no performance implications.

I can probably add another note here explaining all of this?, wdyt?

Reply:
No comment is necessary.  I was curious what the potentially for deduplication was.